software/nosql/src/main/java/brooklyn/entity/nosql/redis/RedisClusterImpl.java

```diff
@@ -20,49 +20,73 @@
 
 import java.util.Collection;
 
+import brooklyn.enricher.Enrichers;
 import brooklyn.entity.Entity;
 import brooklyn.entity.basic.AbstractEntity;
 import brooklyn.entity.basic.Entities;
 import brooklyn.entity.group.DynamicCluster;
 import brooklyn.entity.proxying.EntitySpec;
 import brooklyn.entity.trait.Startable;
+import brooklyn.event.AttributeSensor;
+import brooklyn.event.basic.Sensors;
 import brooklyn.location.Location;
 
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+
 public class RedisClusterImpl extends AbstractEntity implements RedisCluster {
 
-    protected RedisStore master;
-    protected DynamicCluster slaves;
+    private AttributeSensor<RedisStore> MASTER = Sensors.newSensor(RedisStore.class, "redis.master");
+    private AttributeSensor<DynamicCluster> SLAVES = Sensors.newSensor(DynamicCluster.class, "redis.slaves");
 
     public RedisClusterImpl() {
     }
 
     @Override
     public RedisStore getMaster() {
-        return master;
+        return getAttribute(MASTER);
     }
     
     @Override
     public DynamicCluster getSlaves() {
-        return slaves;
+        return getAttribute(SLAVES);
     }
     
     @Override
     public void start(Collection<? extends Location> locations) {
```

Comment:
Also see the code in things like `DynamicCluster.start(locations)`, which does things like `ServiceStateLogic.setExpectedState(this, Lifecycle.STARTING);`. That gives more visibility of what the entity is doing and better error reporting. The finally block also ensures that we'll correctly report failures, and will stop the green-spinning in the web-console that was shown while starting.